src/main/java/vendored/org/apache/dolphinscheduler/common/utils/process/ProcessImplForWin32.java

```diff
@@ -0,0 +1,1048 @@
+                }
+
+                if (ioRedirectSuccess) {
+                    WinBase.PROCESS_INFORMATION pi = new WinBase.PROCESS_INFORMATION();
+                    si.dwFlags = WinBase.STARTF_USESTDHANDLES;
+
+                    boolean createProcSuccess;
+                    String createProcContext;
+                    if (isService.get()) {
+                        createProcContext = "CreateProcessAsUser";
+
+                        final WinBase.SECURITY_ATTRIBUTES threadSa = new WinBase.SECURITY_ATTRIBUTES();
+                        threadSa.lpSecurityDescriptor = null;
+                        threadSa.bInheritHandle = false;
+                        threadSa.write();
+
+                        createProcSuccess = Advapi32.INSTANCE.CreateProcessAsUser(extraInfo.primaryTokenHandle, null, cmd,
+                                extraInfo.processSa, threadSa, true, PROCESS_CREATION_FLAGS,
+                                envblock, path, si, pi);
+                        Kernel32Util.closeHandles(extraInfo.primaryTokenHandle);
+                    } else {
+                        createProcContext = "CreateProcessWithLogonW";
+
+                        WTypes.LPWSTR lpEnvironment = envblock == null ? new WTypes.LPWSTR() : new WTypes.LPWSTR(envblock);
+                        createProcSuccess = Advapi32.INSTANCE.CreateProcessWithLogonW(username, null, password,
+                                Advapi32.LOGON_WITH_PROFILE, null, cmd, PROCESS_CREATION_FLAGS,
+                                lpEnvironment.getPointer(), path, si, pi);
+                    }
+
+                    if (createProcSuccess) {
+                        closeHandle(pi.hThread);
+                        ret = pi.hProcess;
+                        pid = pi.dwProcessId.intValue();
+                    } else {
+                        throw lastErrorProcessCreationException(createProcContext);
+                    }
+                }
+                releaseHolder(ret.getPointer().equals(Pointer.createConstant(0)), pipeError, OFFSET_WRITE);
+                releaseHolder(ret.getPointer().equals(Pointer.createConstant(0)), pipeOut, OFFSET_WRITE);
+            }
+            releaseHolder(ret.getPointer().equals(Pointer.createConstant(0)), pipeIn, OFFSET_READ);
+        }
+        restoreIOEHandleState(stdIOE, inherit);
+        return ret;
+    }
+
+    private synchronized WinNT.HANDLE create(String username,
+                                             String password,
+                                             String cmd,
+                                             final java.util.Map<String,String> envMap,
+                                             final String path,
+                                             final long[] stdHandles,
+                                             final boolean redirectErrorStream) throws ProcessCreationException {
+        // Get and cache process token and isService state
```

Comment:
can you add a comment that this is altered from the original library and why we had to modify it

Each modified file should contain a note that it has been modified